aws/resource_aws_ec2_tag.go

```diff
@@ -11,6 +12,19 @@ import (
 	"github.com/hashicorp/terraform/helper/schema"
 )
 
+type ResourceNotExists struct {
+	message string
+}
+
+func NewResourceNotExists(message string) *ResourceNotExists {
+	return &ResourceNotExists{
+		message: message,
+	}
+}
+func (e *ResourceNotExists) Error() string {
+	return e.message
+}
```

Comment:
Any reason we can't just use a normal `fmt.Errorf`?

```go
return resource.RetryableError(fmt.Errorf("Expected instance to be created but was in state %s", resp.Status))
```

Reply:
This is used later to check the type of error after it exits the retry scope